starter-code/main.js

```diff
@@ -54,8 +55,192 @@ mongoClient.connect(`mongodb://localhost:27017/crunchbase`, (error, db) => {
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+
+            case "7":
+            // 7.- List the 10 companies with more employees sorted ascending (show name and employees).
+            db.collection('companies').find({}, { name: 1, number_of_employees: 1, _id: 0 }).sort({number_of_employees: -1}).limit(10).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result); 
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+
+            case "8":
+            // 8.- Find the company with the name "Facebook"
+            db.collection('companies').find({name: "Facebook"}).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result); 
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+
+            case "9":
+            // 9.- How many employees has Facebook?
+            db.collection('companies').find({name: "Facebook"}, {number_of_employees: 1, _id: 0 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result); 
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+
+            case "10":
+            // 10.- List the name of all the products of Facebook
+            db.collection('companies').find({name: "Facebook"}, {"products.name": 1, _id: 0 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result); 
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+
+            case "11":   
+            // 11.- List the people that are working at Facebook right now (check relationships field)
+            db.collection('companies').find({ $and: [{name: "Facebook"}, { "relationships.is_past": false }] }, {"relationships.person": 1, _id: 0 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result); 
```

Comment:
- Aquí te recomiendo, que una vez hayas obtienes el objeto con la query de la base de datos, apliques JS para filtrar lo que buscas (que `is_past` no sea false).

```suggestion
console.log(result[0].relationships.filter(emp => !emp.is_past).length)
```

Reply:
Jo genial, así me queda mucho más claro, gracias! :pray: